src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -131,15 +166,15 @@ public void OnResponseHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
                         {
                             // Pseudo-headers not allowed in trailers.
                             if (NetEventSource.IsEnabled) _connection.Trace("Pseudo-header in trailer headers.");
-                            throw new HttpRequestException(SR.net_http_invalid_response_pseudo_header_in_trailer);
+                            throw new Http2ProtocolException(SR.net_http_invalid_response_pseudo_header_in_trailer);
                         }
 
                         if (name.SequenceEqual(s_statusHeaderName))
                         {
-                            if (_response != null)
+                            if (_response != null && (int)_response.StatusCode > 199)
```

Comment:
What's the extra condition for?

Reply:
I replaced this with new state as @geoffkizer suggested. The original idea was to enforce only one status line but allow 1xx response as they are not final.